packages/components/src/popover/index.tsx

```diff
@@ -367,12 +408,13 @@ const Popover = (
 	// we need to manually update the floating's position as the reference's owner
 	// document scrolls. Also update the frame offset if the view resizes.
 	useLayoutEffect( () => {
-		const referenceAndFloatingAreInSameDocument =
-			referenceOwnerDocument === document;
-		const hasFrameElement =
-			!! referenceOwnerDocument?.defaultView?.frameElement;
-
-		if ( referenceAndFloatingAreInSameDocument || ! hasFrameElement ) {
+		if (
+			// reference and floating are in the same document
+			referenceOwnerDocument === document ||
+			// the reference's document has a view (i.e. window)
+			// and a frame element (ie. it's an iframe)
+			! referenceOwnerDocument?.defaultView?.frameElement
+		) {
```

Comment:
Runtime change: moved these checks to be inline because TypeScript wouldn't otherwise use them correctly as a typeguard for `defaultView` later in the code